### ISSUE TITLE
PHP upload errors do not trigger error in beforeSave

### ISSUE THREAD

**Issue by curtisgibby** [Xety/Cake3-Upload#8]
You're checking to see if the user didn't upload a file at all, but the other [PHP file-upload errors](http://php.net/manual/en/features.file-upload.errors.php) cause unexpected behavior.

In my case, the user was trying to upload huge files, well beyond the `upload_max_filesize` limit. The "file" would get saved with 0 bytes, the database would get updated, and the system would tell the user that everything was great.

Here's a snippet that I think fixes the issue; just replace [the `UPLOAD_ERR_NO_FILE` check](https://github.com/Xety/Cake3-Upload/blob/a82c46f67adc00645b78e2b0314bda0920dc6d59/src/Model/Behavior/UploadBehavior.php#L70-L72) with something like this:

```php
if (!empty($file['error'])) {
    switch ($file['error']) {
        case UPLOAD_ERR_INI_SIZE:
            $message = "The uploaded file exceeds the upload_max_filesize directive in php.ini : " . ini_get('upload_max_filesize');
            break;
        case UPLOAD_ERR_FORM_SIZE:
            $message = "The uploaded file exceeds the MAX_FILE_SIZE directive that was specified in the HTML form";
            break;
        case UPLOAD_ERR_NO_FILE:
            continue 2; // NOT throwing an exception here
            break;
        case UPLOAD_ERR_PARTIAL:
            $message = "The uploaded file was only partially uploaded";
            break;
        case UPLOAD_ERR_NO_TMP_DIR:
            $message = "Missing a temporary folder";
            break;
        case UPLOAD_ERR_CANT_WRITE:
            $message = "Failed to write file to disk";
            break;
        case UPLOAD_ERR_EXTENSION:
            $message = "File upload stopped by extension";
            break;
        default:
            $message = "Unknown upload error";
            break;
    }

    throw new \ErrorException($message);
}
```

**Comment by Xety** [comment]
Hello,

Sorry i was busy since a while now and i didn't check my github for 2 months.

Indeed, i missed to check all the file-upload errors possibilities. But instead of doing this check in the `beforeSave`, it would be better to do a new protected function. Can you please do a PR with the changes ? If you don't have the time, i will try to do it as soon as i can.

Thanks for the report.

**Comment by Xety** [comment]
Closed as #10 is open now.
